FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/CustomSchemaConverter.cs

```diff
@@ -57,10 +57,19 @@ public class CustomSchemaConverter : IDataSchemaConverter
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            if (string.IsNullOrWhiteSpace(_schemaImageReference))
+            {
+                _logger.LogError($"Schema image reference is empty or null.");
+                throw new ParquetDataProcessorException($"Schema image reference is empty or null.");
+            }
+
             if (_templateProvider == null)
             {
-                _logger.LogError($"No valid template provider be found, maybe the schema image reference is empty or null.");
-                throw new ParquetDataProcessorException($"No valid template provider be found, maybe the schema image reference is empty or null.");
+                var templateCollections = _containerRegistryTemplateProvider.GetTemplateCollectionAsync(
```

Comment:
Maybe we can add a lock here to avoid multiple requests?

Reply:
Fixed, thanks.